### PR TITLE
Lua_api.txt: Improve bullet point indentation consistency

### PR DESCRIPTION
**Pull request by Ezhh**
As discussed and agreed in #6910. 

This fixes bullet point spacing inconsistencies throughout lua_api.txt.

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -1304,7 +1304,7 @@ in the form of a table.  This table specifies the following fields:
   * `prob` (alias `param1`): the probability of this node being placed (default: 255)
   * `param2`: the raw param2 value of the node being placed onto the map (default: 0)
   * `force_place`: boolean representing if the node should forcibly overwrite any
-     previous contents (default: false)
+    previous contents (default: false)
```
Comment (paramat) [inline doc/lua_api.txt]: Lines 1303-1307 seem to need 2 more spaces of indent?

Comment (Ezhh) [reply doc/lua_api.txt]: Nice catch - will update

Comment (paramat) [reply doc/lua_api.txt]: Not sure, not essential.

Comment (Ezhh) [reply doc/lua_api.txt]: It should be done for consistency with other second level bullets, so I'd like to.

Comment (Ezhh) [reply doc/lua_api.txt]: Done.

Comment (paramat) [reply doc/lua_api.txt]: It's ok as it is.

Comment (Ezhh) [reply doc/lua_api.txt]: Now that I changed it it is. It was inconsistent.

Comment (paramat) [reply doc/lua_api.txt]: Sorry just saw your comments, the new PR is fine.

**Comment by paramat** [comment]
So might you want to indent lines 598-601? 609-612?

**Comment by Ezhh** [comment]
Yes, and done. Scanning through 5k+ lines leaves it easy to miss a few. :)

**Comment by paramat** [comment]
Lines 2880-2887 might need attention.
2880 less indent.
2881-2887 more indent.

**Comment by paramat** [comment]
3339 and 3341
3564-3567

**Comment by paramat** [comment]
4196-4199

**Comment by paramat** [comment]
Otherwise good, finished review.

**Comment by Ezhh** [comment]
For 2880-2887, it looks like lines that have "-" might actually need further indentation. Also that the whole block from 2880-2887 might need another level before considering lines with "-". I've sort of skipped over bits like this because it's getting into full on content review to make these judgement calls, instead of just high level spacing clean ups.

**Comment by paramat** [comment]
Ok you can leave 2880-2887 to later if you want.

**Comment by Ezhh** [comment]
What is the issue with 4196-4199?

**Comment by paramat** [comment]
They should have 2 spaces of indent? Not sure, looking.

**Comment by paramat** [comment]
3382

**Comment by paramat** [comment]
I mean 3383

**Comment by Ezhh** [comment]
Added extra level of indent for 3339 and 3340.

Indents increased for 3563-3566 and for 3569

**Comment by paramat** [comment]
4196-4199 is debateable so i don't mind either way.

**Comment by Ezhh** [comment]
4196-4199 should actually have no indent instead.

**Comment by paramat** [comment]
Agreed.

**Comment by paramat** [comment]
Unless i see more review is done.

**Comment by Ezhh** [comment]
4196-4199 indentation removed.

Indented 3382. (It is 3382, not 3383 that seems wrong, though it did make for confusing reading.)

**Comment by paramat** [comment]
Sorry my line numbers are of MT master.

**Comment by Ezhh** [comment]
No worries, worked out what you were using :)

Seeing if I can figure out 2880-2887 anyway now.

Have I missed any of the others you pointed out?

**Comment by paramat** [comment]
:+1:

**Comment by paramat** [comment]
```
* `minetest.emerge_area(pos1, pos2, [callback], [param])`
    * Queue all blocks in the area from `pos1` to `pos2`, inclusive, to be asynchronously
    * fetched from memory, loaded from disk, or if inexistent, generates them.
    * If `callback` is a valid Lua function, this will be called for each block emerged.
    * The function signature of callback is:
      `function EmergeAreaCallback(blockpos, action, calls_remaining, param)`
        * `blockpos` is the *block* coordinates of the block that had been emerged
        * `action` could be one of the following constant values:
          `minetest.EMERGE_CANCELLED`, `minetest.EMERGE_ERRORED`, `minetest.EMERGE_FROM_MEMORY`,
          `minetest.EMERGE_FROM_DISK`, `minetest.EMERGE_GENERATED`
        * `calls_remaining` is the number of callbacks to be expected after this one
        * `param` is the user-defined parameter passed to emerge_area (or nil if the
          parameter was absent)
```

**Comment by paramat** [comment]
Removed the asterisk from 2877

**Comment by Ezhh** [comment]
Updated for 2880-2887 (please review as I didn't see your version until just now).

**Comment by paramat** [comment]
I don't think 2880 is a bullet point, it's more a continuation of the previous line.
I prefer my suggestion.

**Comment by paramat** [comment]
2883-2884 each of those should be a point, or none.

**Comment by Ezhh** [comment]
It follows a colon on the previous line, and that would generally mean a bullet.

**Comment by paramat** [comment]
Ok.

**Comment by paramat** [comment]
`+        *   `function EmergeAreaCallback(blockpos, action, calls_remaining, param)`
Has 3 spaces after the asterisk.

**Comment by Ezhh** [comment]
Removed two of them (sorry, missed it when straightening up the rest)

**Comment by paramat** [comment]
2883, 2884 each of those 5 a point?

**Comment by Ezhh** [comment]
Hmm, those should technically be five points, but have been stuffed onto two lines. I don't really want to make it five though, so let me re-word it a little.

**Comment by paramat** [comment]
Those are fine as line continuations too (none as points).

**Comment by Ezhh** [comment]
Yes... I rewrote to one line first, and actually didn't like it, so five it is.

**Comment by paramat** [comment]
All good for me :+1:

**Comment by paramat** [comment]
Oh 2877 isn't a bullet.

**Comment by Ezhh** [comment]
You are getting into content review now as opposed to formatting now. If you keep going, you will never ever stop finding issues.

**Comment by paramat** [comment]
That's all :+1:

**Comment by paramat** [comment]
5435b07d4e4e0158b75d9432bb8971cef40fd007

**Pull request closed without merging**